packages/siwe/src/client.ts

```diff
@@ -0,0 +1,82 @@
+import type {
+  CreateSiweMessageArgs,
+  VerifySiweMessageArgs,
+  SiweControllerClient,
+  PublicStateControllerState,
+  SiweControllerClientState
+} from '@web3modal/core'
+import { Web3ModalScaffold, type ScaffoldOptions } from '@web3modal/scaffold'
+
+// -- Types ---------------------------------------------------------------------
+export interface Web3ModalSiweClientOptions extends ScaffoldOptions {
+  siweConfig: SiweControllerClient
+}
+
+export type Web3ModalState = PublicStateControllerState & SiweControllerClientState
+
+// -- Client --------------------------------------------------------------------
+export class Web3ModalSiwe extends Web3ModalScaffold {
```

Comment:
This one is interesting, siwe client should be it's own unique thing and not extend Web3ModalScaffold package. Instead siweClient will be passed to scaffold package as an option (of wagmi or ethers packages).